test_libs/pyspec/tests/phase0/conftest.py

```diff
@@ -3,34 +3,31 @@
 from eth2spec.phase0 import spec
 from preset_loader import loader
 
-from .helpers import (
-    create_genesis_state,
-)
-
+from tests.phase0 import helpers
+from tests.phase0 import test_finality
 
 def pytest_addoption(parser):
     parser.addoption(
         "--config", action="store", default="minimal", help="config: make the pyspec use the specified configuration"
     )
 
-
 @pytest.fixture(autouse=True)
 def config(request):
     config_name = request.config.getoption("--config")
     presets = loader.load_presets('../../configs/', config_name)
     spec.apply_constants_preset(presets)
-
+    helpers.spec = spec
+    request.function.__globals__['spec'] = spec
```

Comment:
So this is my hack to get the spec version available to the tests to be the correct version independent of whether it is the phase 0 or phase 1 spec. It is not the prettiest, but it is the best solution I could come up with. It allows phase 0 tests to be imported into phase 1 and the tests to be re-run against the phase1 spec with the phase 1 spec objects in the test `helpers.py`. If someone finds a better solution, I am keen to hear about it.

Reply:
The `helpers.py` hack here isn't really sustainable. In #1102, we are dividing out helpers into better organized files.

I'll look into this a bit more tomorrow

Reply:
To be clear, is your objection to the use of a monolithic `helpers.py`, to the particular method by which I inject the correct version of `spec.py` into `helpers.py` or to the use of `request.function.__globals__['spec'] = spec` and `request.function.__globals__['helpers'] = helpers` to provide the spec and helpers to the tests?

I agree with the direction taken in #1102 w.r.t. breaking up `helpers.py`. I would prefer an alternative to my hack use of pytest globals for spec injection, but I am unable to come up with one myself. I think it is very important that the spec version used by the helpers is the same as the spec version being tested.

Reply:
I actually think this hack is ok, if there is no nice way to provide this in pytest :) I actually tried to do something like this before but failed because I was still using normal "import"s